clang/lib/Tooling/DependencyScanning/DependencyScanningFilesystem.cpp

```diff
@@ -201,11 +201,8 @@ const CachedRealPath &DependencyScanningFilesystemSharedCache::CacheShard::
   return *StoredRealPath;
 }
 
-static bool shouldCacheStatFailures(StringRef Filename) {
-  StringRef Ext = llvm::sys::path::extension(Filename);
-  if (Ext.empty())
-    return false; // This may be the module cache directory.
-  return true;
+bool DependencyScanningWorkerFilesystem::shouldBypass(StringRef Path) const {
+  return BypassedPathPrefix && Path.starts_with(*BypassedPathPrefix);
```

Comment:
Do we always expect canonical paths here? Are paths always rendered using the style of the target platform?

Reply:
I thought that the rest of the compiler just appends to the `Path` we pass in here. But looking into that a bit closer I see that `pruneModuleCache()` in `CompilerInstance.cpp` calls `llvm::sys::path::native()` on this and `HeaderSearch::getCachedModuleFileNameImpl()` calls `llvm::sys::fs::make_absolute()`. I'll investigate if these are necessary and if so, I'll try to unify things so that just doing string prefix match is enough in all situations.

Reply:
`ParseHeaderSearchArgs()` in `CompilerInvocation.cpp` already calls `llvm::sys::fs::make_absolute()` with either the process CWD or `-working-directory`, so the call to the same function in `HeaderSearch::getCachedModuleFileNameImpl()` seems to be redundant for Clang instances that were constructed from command line.

Reply:
It doesn't canonicalize if it's already absolute, which is commonly the case for the module cache dir. If we're going to canonicalize on each access anyway, I think it's fine to canonicalize in CompilerInstance at the start.

Reply:
https://github.com/llvm/llvm-project/pull/102540